docs/methodology/timeseries_imputation.rst

```diff
@@ -0,0 +1,164 @@
+detection and missing value imputation on some of the timeseries data we publish, while
+also providing access to the originally reported values. The infrastructure for this
+work lives in the :mod:`pudl.analysis.timeseries_cleaning` module. Thus far we have only
+applied these methods to the :doc:`/data_sources/eia930` and
+:doc:`/data_sources/ferc714` hourly electricity demand data.
+
+For both of these demand timeseries, we flag anomalous values for imputation using
+heuristics developed and maintained by:
+
+- `Tyler Ruggles <https://github.com/truggles>`__
+- `Alicia Wongel <https://github.com/awongel>`__
+- `Greg Schivley <https://github.com/gschivley>`__
+- `David Farnham <https://github.com/d-farnham>`__
+
+Their anomaly detection heuristics were originally implemented in
+`the truggles GitHub repo, "EIA Cleaned Hourly Electricity Demand Code,"
+<https://github.com/truggles/EIA_Cleaned_Hourly_Electricity_Demand_Code>`__ (also
+`archived on Zenodo <http://doi.org/10.5281/zenodo.3737085>`__) and published in
+`Developing reliable hourly electricity demand data through screening and imputation
+<https://doi.org/10.1038/s41597-020-0483-x>`__.
+
+This work was originally applied only to the EIA-930 hourly electricity demand data
+reported by balancing authorities (BAs). We re-implemented the heuristics in PUDL and
+applied them to the very similar FERC-714 demand data reported by electricity planning
+areas, which has a much longer history (going back to 2006, vs. 2015 for the EIA-930).
+
+We also adopted a different imputation method designed for correlated timeseries that
+display periodicity. It is very computationally efficient, allowing timeseries
+with tens of millions of values to be imputed in a few minutes on a laptop. The
+algorithm was designed for multivariate time series forecasting and we adapted it
+from code published by: `Xinyu Chen <https://xinychen.github.io/>`__ in their `Tensor
+decomposition for machine learning repository
+<https://github.com/xinychen/tensor-learning>`__. The method is described in more detail
+in the following papers:
+
+- `Low-Rank Autoregressive Tensor Completion for Multivariate Time Series Forecasting <https://arxiv.org/abs/2006.10436>`__
+- `Scalable Low-Rank Tensor Learning for Spatiotemporal Traffic Data Imputation <https://arxiv.org/abs/2008.03194>`__
+
+This method could potentially be applied to other instances of correlated periodic data.
+For example, we've considered applying these methods to EIA 930 hourly net generation.
+
+How it Works
+~~~~~~~~~~~~
+
+First we identify any values which are missing or deemed anomalous using the heuristics
+developed by Ruggels et al., and then we impute those "flagged" values. You can see the
+list of reasons why a value might be flagged for imputation in the
+:ref:`core_pudl__codes_imputation_reasons` table. After we've flagged the missing and
+anomalous values, we impute each year of data independently. The imputation algorithm
```

Comment:
Should this be "we impute flagged values within each year of data independently"?

Reply:
Yeah that's probably better.